core/src/main/java/jenkins/security/ResourceDomainConfiguration.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2019 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.security;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.Util;
+import hudson.util.FormValidation;
+import jenkins.diagnostics.RootUrlNotSetMonitor;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.GlobalConfigurationCategory;
+import jenkins.model.JenkinsLocationConfiguration;
+import jenkins.util.UrlHelper;
+import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.annotation.Nonnull;
+import javax.servlet.http.HttpServletRequest;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Configure the resource root URL, an alternative root URL to serve resources from
+ * to not need Content-Security-Policy headers, which mess with desired complex output.
+ *
+ * @see ResourceDomainFilter
+ * @see ResourceDomainRootAction
+ *
+ * @since TODO
+ */
+@Extension(ordinal = JenkinsLocationConfiguration.ORDINAL-1) // sort just below the regular location config
+@Restricted(NoExternalUse.class)
+@Symbol("resourceDomain")
+public class ResourceDomainConfiguration extends GlobalConfiguration {
+
+    private String resourceRootUrl;
```

Comment:
the configuration exports fine, I'm a little torn on whether this is repeating information in the key from it's parent or whether it's needed

![image](https://user-images.githubusercontent.com/21194782/65986218-56ce9400-e47b-11e9-836f-9fdcf6bc7a85.png)

Reply:
@timja Addressed in https://github.com/jenkinsci/jenkins/pull/4239/commits/d0d37ab6c8121d4d4427c3359b2933d0d04dfbb5

Untested, but I expect:

```
resourceRoot:
  url: "http://build-resources.acme.org/"
```